### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Warhammer4000**
@coderabbitai could you please provide me with a proper changelog for this PR?

@coderabbitai could you please provide me with a proper readme for this PR?

## Summary by CodeRabbit

- **New Features**
	- Introduced a new save system with enhanced encryption, archiving, and multiple slot management.
	- Added a save system prefab for easy integration into projects.
	- Implemented asynchronous save and load operations for improved performance.

- **Enhancements**
	- Improved cloud save functionality and added configuration options for save paths and slot management.
	- Integrated Unity's UniTask for efficient asynchronous operations and added support for file archiving using SharpZipLib utility.

- **Bug Fixes**
	- Updated GUIDs for assets to ensure correct references.

- **Documentation**
	- Provided samples and demo scenes to showcase save system usage.

- **Platform Support**
	- Added new settings for Xbox Series S/X and Xbox One deployment.

- **Refactor**
	- Restructured save system installation and encryption logic for simplicity and reliability.

- **Upgrades**
	- Updated Unity editor version to enhance compatibility and performance.

### CODE REVIEW

**Comment by Warhammer4000** [comment]
@coderabbitai could you please provide me with a proper changelog for this PR? from all commits

**Comment by Warhammer4000** [comment]
@coderabbitai could you please provide me with a proper readme for this PR from all commits

**Comment by Warhammer4000** [comment]
> @Warhammer4000 The changelog for this PR, based on all commits, is as follows:
> 
> ```
> 71f3fa0 - v2.0.0 (Tanimul Haque Khan, 42 minutes ago)
> 0f2b6d3 - v2.0.0 (Tanimul Haque Khan, 23 hours ago)
> ```
> 
> It appears that there are two commits both marked as version 2.0.0. If you need more detailed information about the changes in each commit, I can provide that as well.
> 
> Analysis chain
> Scripts executed

Please provide a detailed changelog

**Comment by Warhammer4000** [comment]
@coderabbitai please provide a detailed changelog

**Comment by Warhammer4000** [comment]
@coderabbitai generate changelog by comparing code from main and this pull request. don't use commit message as changelog. Instead use file compare.

**Comment by Warhammer4000** [comment]
@coderabbitai please provide further details on specific changes within the files.

File: Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                Directory.CreateDirectory(slotPath);
+
+            }
+        }
+
+        internal async UniTask<SaveSlot> GetSaveSlotMetaData(int index)
+        {
+            SaveSlot slotMeta=new SaveSlot($"Save Slot {index}");
+
+            string slotMetaPath = Path.Combine(_slotConfiguration.SavePathRoot, $"Save Slot {index}", _slotConfiguration._slotMetafileName);
```
Comment (gitOpu) [inline Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs]: $"Save Slot {index}"
white space may be problematic on different platforms. it's better to remove white space from the path or folder name

File: Assets/Packages/com.studio23.ss2.savesystem/Runtime/Data/SlotConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+
+    public string _slotDatafolderName = "Data";
+    public string _slotDataBackupFileName = "Data.bak";
+    public string _slotMetafileName = "data.meta";
+
+    public string saveFileExtention = ".taz";
+
+
+    public string _saveRootFolderName = "SaveData";
```
Comment (gitOpu) [inline Assets/Packages/com.studio23.ss2.savesystem/Runtime/Data/SlotConfiguration.cs]: public variable should start in the upper case

File: Assets/Packages/com.studio23.ss2.savesystem/Samples/Demo/Scripts/ItemDataBehaviour.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
+using Studio23.SS2.SaveSystem.Interfaces;
+using UnityEngine;
+
+
+public class ItemDataBehaviour : MonoBehaviour, ISaveable
+{
+
+    public string _uniqueID;
```
Comment (gitOpu) [inline Assets/Packages/com.studio23.ss2.savesystem/Samples/Demo/Scripts/ItemDataBehaviour.cs:8]: there should be instructions as [Title], for creating uniqueID,

**Review by gitOpu — APPROVED** [review]

File: Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+
+    [RequireComponent(typeof(FileProcessor))]
+    internal class SaveSlotProcessor : MonoBehaviour
+    {
+
+        [SerializeField] internal SaveSlot _selectedSlot;
+
+        [SerializeField] internal SlotConfiguration _slotConfiguration;
+        [SerializeField] internal FileProcessor _fileProcessor;
+        [SerializeField] internal ArchiverBase archiverBase;
```
Comment (BDeshiDev) [inline Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs]: Naming convention underscore

File: Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs (hunk 1 above)
Comment (Warhammer4000) [reply Assets/Packages/com.studio23.ss2.savesystem/Runtime/Core/SaveSlotProcessor.cs]: Fixed

File: Assets/Packages/com.studio23.ss2.savesystem/Runtime/Data/SlotConfiguration.cs (hunk 2 above)
Comment (Warhammer4000) [reply Assets/Packages/com.studio23.ss2.savesystem/Runtime/Data/SlotConfiguration.cs]: Done

File: Assets/Packages/com.studio23.ss2.savesystem/Samples/Demo/Scripts/ItemDataBehaviour.cs (hunk 3 above)
Comment (Warhammer4000) [reply Assets/Packages/com.studio23.ss2.savesystem/Samples/Demo/Scripts/ItemDataBehaviour.cs:8]: it's from sample.

File: Assets/Packages/com.studio23.ss2.savesystem/Runtime/Data/SlotConfiguration.cs [hunk 5]
```diff
@@ -1,23 +1,26 @@
...
+namespace Studio23.SS2.SaveSystem.Data
 {
-    [SerializeField] public int _slotCount = 10;
+    [CreateAssetMenu(fileName = "SlotConfig", menuName = "Studio-23/SaveSystem/Config", order = 1)]
+    public class SlotConfiguration : ScriptableObject
+    {
+        [SerializeField] public int SlotCount = 10;
 
-    [SerializeField] public bool _enableBackups = true;
+        [SerializeField] public bool EnableBackups = true;
```
Comment (BDeshiDev) [inline Assets/Packages/com.studio23.ss2.savesystem/Runtime/Data/SlotConfiguration.cs]: [SerializeField] not necessary

File: Assets/Packages/com.studio23.ss2.savesystem/Editor/Scripts/SaveSlotProcessorEditor.cs [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+using Studio23.SS2.SaveSystem.Data;
+using UnityEditor;
+
+
+namespace Studio23.SS2.SaveSystem.Editor
+{
+    [CustomEditor(typeof(SaveSlotProcessor))]
+    public class SaveSlotProcessorEditor : UnityEditor.Editor
+    {
+        private bool showSlotConfig = true;
```
Comment (BDeshiDev) [inline Assets/Packages/com.studio23.ss2.savesystem/Editor/Scripts/SaveSlotProcessorEditor.cs:11]: Naming convention.

File: Assets/Packages/com.studio23.ss2.savesystem/Editor/Scripts/SaveSystemEditor.cs [hunk 7]
```diff
@@ -1,21 +1,20 @@
...
 
-namespace Studio23.SS2.SaveSystem.Core
+namespace Studio23.SS2.SaveSystem.Editor
 {
-    [CustomEditor(typeof(SaveSystem))]
+    [CustomEditor(typeof(Core.SaveSystem))]
     public class SaveSystemEditor : UnityEditor.Editor
     {
         private int slotIndex = 0; // Default slot index
         private bool showDebugTools = true;
```
Comment (BDeshiDev) [inline Assets/Packages/com.studio23.ss2.savesystem/Editor/Scripts/SaveSystemEditor.cs]: naming convention

**Review by BDeshiDev — COMMENTED** [review]
Minor var name convention changes

**Review by gitOpu — APPROVED** [review]

**Review by zubEpty — APPROVED** [review]

**Pull request merged**